### PR TITLE
--deploy-group option for paasta status

### PR DESCRIPTION
**Pull request by oktopuz**
internal ticket: PAASTA-9596

### CODE REVIEW

File: paasta_tools/cli/cmds/status.py [hunk 1]
```diff
@@ -285,36 +295,49 @@ def report_status(service, deploy_pipeline, actual_deployments, cluster_whitelis
     return 0 if all([return_code == 0 for return_code in return_codes]) else 1
 
 
+def paasta_args_mixer(args, service):
+    clusters_instances = get_cluster_instance_map_for_service(soa_dir=args.soa_dir, service=service,
+                                                              deploy_group=args.deploy_group) \
+        if args.deploy_group is not None else {}
```
Comment (oktopuz) [inline paasta_tools/cli/cmds/status.py]: fixed.

File: paasta_tools/cli/cmds/status.py [hunk 2]
```diff
@@ -78,6 +81,13 @@ def add_subparser(subparsers):
              "For example: --instances canary,main"
     )  # No completer because we need to know service first and we can't until some other stuff has happened
     status_parser.add_argument(
+        '-l', '--deploy-group',
+        help=('Name of the deploy group which you want to get status for. This option is intended to be used '
+              'without --clusters (-c)  and --instances (-i). '
+              'If both --deploy-group and --instances (or --clusters) are provided, only --instances (or --clusters) '
```
Comment (oktopuz) [inline paasta_tools/cli/cmds/status.py]: This is implemented.

**Comment by oktopuz** [comment]
@EvanKrall, could you review this PR?

**Comment by solarkennedy** [comment]
This mixer is CGN but ship!

**Comment by EvanKrall** [comment]
When I said to add a deploy_group_whitelist option to report_status, I was thinking something more like this: https://github.com/Yelp/paasta/commit/c24b09c5e6029b7d25068c7715aca55355e05f95

**Pull request merged**